lib/crash_reporter.js

```diff
@@ -0,0 +1,136 @@
+var logs = require('./logs');
+var os = require('os');
+var packageJson = require('../package.json');
+var preferences = require('./preferences');
+var request = require('request');
+
+// the value of the crash reporter preference
+// the value has to be one of 'on' or 'off'
+var CRASH_REPORTER_PREFERENCE = 'crash.reporter.preference';
+var SUBMIT_CRASH_URL = 'http://tessel-error-reporter.appspot.com/crashes/submit';
+
+// override for testing
+if (process.env.DEV_MODE === 'true') {
+  SUBMIT_CRASH_URL = 'http://localhost:9080/crashes/submit';
+}
+
+// the only instance of the crash reporter
+var crashReporter = null;
+
+var CrashReporter = function() {
+  this.preferences = new preferences.Preferences();
+};
+
+CrashReporter.getInstance = function() {
+  if (crashReporter) {
+    return crashReporter;
+  } else {
+    crashReporter = new CrashReporter();
+    return crashReporter;
+  }
+};
+
+CrashReporter.prototype.turnOff = function() {
+  return this.preferences
+    .write(CRASH_REPORTER_PREFERENCE, 'off')
+    .catch(function(error) {
+      // do nothing
+      // do not crash the crash reporter :)
+      logs.err('Error turning off crash reporter preferences', error);
+    });
+};
+
+CrashReporter.prototype.turnOn = function() {
+  return this.preferences
+    .write(CRASH_REPORTER_PREFERENCE, 'on')
+    .catch(function(error) {
+      // do nothing
+      // do not crash the crash reporter :)
+      logs.err('Error turning on crash reporter preferences', error);
+    });
+};
+
+CrashReporter.prototype.submit = function(report) {
+  var that = this;
+  that.preferences.read(CRASH_REPORTER_PREFERENCE, 'on')
+    .then(function(value) {
+      if (value === 'on') {
+        var labels = [
+          // tessel cli labels
+          packageJson.name,
+          `CLI version: ${packageJson.version}`,
+          // node version
+          `Node version: ${process.version}`,
+          // os version
+          `OS platform: ${os.platform()}`,
+          `OS release: ${os.release()}`
+        ];
+
```

Comment:
This can just be one multi line template string. Use `tags.stripIndent` (add "common-tags" to requires in this file)